DQMOffline/Trigger/plugins/MssmHbbBtagTriggerMonitor.cc

```diff
@@ -0,0 +1,175 @@
+#include <math.h>
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "DQM/TrackingMonitor/interface/GetLumi.h"
+#include "CommonTools/TriggerUtils/interface/GenericTriggerEventFlag.h"
+
+#include "DataFormats/Common/interface/Handle.h"
+
+
+#include "DQMOffline/Trigger/plugins/MssmHbbBtagTriggerMonitor.h"
+
+#include "TLorentzVector.h"
+
+
+// -----------------------------
+//  constructors and destructor
+// -----------------------------
+
+MssmHbbBtagTriggerMonitor::MssmHbbBtagTriggerMonitor( const edm::ParameterSet& iConfig )
+{
+  folderName_             = iConfig.getParameter<std::string>("dirname");
+  processname_            = iConfig.getParameter<std::string>("processname");
+  pathname_               = iConfig.getParameter<std::string>("pathname");
+  triggerobjbtag_         = iConfig.getParameter<std::string>("triggerobjbtag");
+  triggerSummaryLabel_    = iConfig.getParameter<edm::InputTag>("triggerSummary");
+  triggerResultsLabel_    = iConfig.getParameter<edm::InputTag>("triggerResults");
+  triggerSummaryToken_    = consumes <trigger::TriggerEvent> (triggerSummaryLabel_);
+  triggerResultsToken_    = consumes <edm::TriggerResults>   (triggerResultsLabel_);
+  offlineCSVPFToken_      = consumes<reco::JetTagCollection> (iConfig.getParameter<edm::InputTag>("offlineCSVPF"));
+
+}
+
+MssmHbbBtagTriggerMonitor::~MssmHbbBtagTriggerMonitor()
+{
+
+}
+
+void MssmHbbBtagTriggerMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
+				 edm::Run const        & iRun,
+				 edm::EventSetup const & iSetup) 
+{  
+   
+  std::string currentFolder = folderName_ ;
+  ibooker.setCurrentFolder(currentFolder.c_str());
+  
+  pt_jet1_ = ibooker.book1D("pt_jet1","pt_jet1",20,0,200);
+  pt_jet2_ = ibooker.book1D("pt_jet2","pt_jet2",20,0,200);
+  
+  pt_probe_ = ibooker.book1D("pt_probe","pt_probe",20,0,200);
+  pt_probe_match_ = ibooker.book1D("pt_probe_match","pt_probe_match",20,0,200);
+  
+
+  discr_offline_btagcsv_jet1_ = ibooker.book1D("discr_offline_btagcsv_jet1","discr_offline_btagcsv_jet1",20,0,10);
+  discr_offline_btagcsv_jet2_ = ibooker.book1D("discr_offline_btagcsv_jet2","discr_offline_btagcsv_jet2",20,0,10);
+  
+  // Initialize the GenericTriggerEventFlag
```

Comment:
why are you not using the GenericTriggerEventFlag ?
it can be used for selecting events based on
- DCS flag (this is very important for jobs run @tier0, because you do not have any json ...)
- HLT and L1 results (those can be even configured by global tag !)